### PR TITLE
Add support for notices and chat subject

### PR DESCRIPTION
**Pull request by ghost**
To mirror the UI specified in the task OC-2654 description, two new features were added:

1. Displaying notices on a step: This can be used to send the user some sort of report, not in the form of a message.
2. Set a subject for the chat: If no subject is set, it is not displayed.

Some CSS changes were applied here, as I believe they are reasonable defaults to be used by anyone.

**Testing instructions**

1. Setup devstack on `ficus.master`
2. Checkout this branch: `git clone -b lucas/OC-2654 https://github.com/snowcrshd/xblock-chat`
3. Log into your devstack system and install the `xblock-chat` to your edx-platform environment.
4. Open Studio, create a unit with the chat xblock (_use the default steps_)
5. Edit the chat xblock, by adding a `subject` to the chat. **Check that subject is rendered on top of chat**
6. Walk through the chat's questions and responses. **Check that a notice is displayed after being asked if you would like to learn more about the xblock**

**Screenshots**

screen shot 2017-06-11 at 21 09 35

screen shot 2017-06-11 at 21 09 15

EDIT: fixed screenshots

### CODE REVIEW

File: chat/chat.py
Comment (mtyaka) [inline chat/chat.py]: I would prefer to keep subject empty by default.

File: chat/chat.py
Comment (mtyaka) [inline chat/chat.py:60]: Please add a test for this. It doesn't have to be anything fancy, just checking that if you set subject to a non-blank string, the `.subject` div is present and it contains the subject, and if the subject is blank, the `.subject` div is not present at all.

File: chat/default_data.py
Comment (mtyaka) [inline chat/default_data.py:36]: Please add a test case for notices, and add some info about notices to the README.

File: chat/public/css/chat.css
Comment (mtyaka) [inline chat/public/css/chat.css]: I think it would look better if the top padding was the same as bottom padding. It would make the test look vertically centered, just like the icon. I think 15 or 20px for both top and bottom paddings would be fine. I would also increase the left padding to 40px and move the icon a couple of pixels to the right.

File: chat/public/css/chat.css
Comment (mtyaka) [inline chat/public/css/chat.css]: Is `z-index` required here? Things seem to work correctly if I remove it.

File: chat/public/js/src/chat.js
Comment (mtyaka) [inline chat/public/js/src/chat.js]: We currently don't have a formal JS style guide, but please always use curly braces in if/else blocks.

File: chat/public/js/src/chat.js
Comment (mtyaka) [inline chat/public/js/src/chat.js]: Please add curly braces.

**Review by mtyaka — CHANGES_REQUESTED** [review]
I requested some minor changes and tests, but otherwise this looks good. Nice work @snowcrshd!

File: chat/public/js/src/chat.js
Comment (ghost) [reply chat/public/js/src/chat.js]: Will fix this!

File: chat/public/js/src/chat.js
Comment (ghost) [reply chat/public/js/src/chat.js]: Will fix this!

File: chat/public/css/chat.css
Comment (ghost) [reply chat/public/css/chat.css]: Hmm, yeah, your're right.

I just added this to be sure it would be on top, but didn't check if it was actually needed.

I'll remove it from the .css

File: chat/public/css/chat.css
Comment (ghost) [reply chat/public/css/chat.css]: Good idea. I'll edit the css and update the PR.

Thanks :)

**Comment by ghost** [comment]
@mtyaka I applied the changes you suggested.

When we are good to merge, I'll squash the commits.

File: README.md
Comment (mtyaka) [inline README.md:130]: Regular users don't usually add custom CSS to their LMS, though. Can you mention that  we currently have built in styles for `correct` and `incorrect` notice types?

File: tests/integration/test_chat.py
Comment (mtyaka) [inline tests/integration/test_chat.py]: Nit: Since `{}` is valid yaml, I got confused for a while why you'd want to set the notice type to an empty dict :) Named placeholder (`{notice_type}`) would make it more obvious that it's a placeholder.

File: tests/integration/test_chat.py
Comment (mtyaka) [inline tests/integration/test_chat.py:937]: Can you also verify that `subject_element.text == chat_subject`?

File: tests/integration/test_chat.py
Comment (mtyaka) [inline tests/integration/test_chat.py:952]: Can we also verify that `notice_element.text == "Yay! Notices are working."`?

**Review by mtyaka — APPROVED** [review]
Thanks for the updates @snowcrshd, this is looking great! I added a few more minor comments. 

You opened this PR from your own fork, which is why CircleCI didn't automatically run the test suite.
Once you address the comments, can you please squash the commits, push the squashed commit to a new branch on this repository, and then open a new PR? That should trigger the CircleCI build. Once the build is green, we can merge this :)

I think you should have sufficient permissions to push to this repository, but let me know if you don't.

File: README.md
Comment (ghost) [reply README.md:130]: I had not considered this, good point. I added a mention to the supported types.

File: tests/integration/test_chat.py
Comment (ghost) [reply tests/integration/test_chat.py]: Added the placeholder. Also changed `notice-text` to hold a placeholder, to make it simpler to write tests.

File: tests/integration/test_chat.py
Comment (ghost) [reply tests/integration/test_chat.py:937]: Added the check. I used `self.assertTrue(chat_subject in subject_element[0].text)`, to make the test more resilient to any changes in the way we display the contents.

File: tests/integration/test_chat.py
Comment (ghost) [reply tests/integration/test_chat.py:952]: Added the check. Did the same thing I mentioned for the subject test.

**Comment by ghost** [comment]
Request changes applied.

Unfortunately, I don't have push permission to this repository. @mtyaka could you help me out on that?

**Comment by mtyaka** [comment]
@snowcrshd Can you try again? I think I got the permissions right this time.

**Comment by ghost** [comment]
@mtyaka it worked now. CircleCI is running https://github.com/open-craft/xblock-chat/pull/8

I'm closing this PR in favor of #8

**Pull request closed without merging**
